test/Base_Tests/src/Data/Numbers_Spec.enso

```diff
@@ -591,21 +587,25 @@ add_specs suite_builder =
             v2 = (Complex.new 6 6)
             Ordering.compare v1 v2 . catch Incomparable_Values (_->42) . should_equal 42
 
-        group_builder.specify "Equality of complex and number" pending=Complex.pending_equality <|
+        group_builder.specify "Equality of complex and number" <|
             v1 = (Complex.new 3)
             v2 = 3
+            r1 = v1==v2
+            r2 = v2==v1
+            r1 . should_be_true
+            r2 . should_be_true
```

Comment:
@GregoryTravis, these tests are now passing with:
```
sbt:enso> runEngineDistribution --run test/Base_Tests Number
Number Conversions: [9/9, 130ms]
    - Complex plus Integer [24ms]
    - Integer plus Complex [12ms]
    - Equality of complex and complex [9ms]
    - Inequality of complex and complex [21ms]
    - Equality of complex and number [8ms]
    - Equality of number and complex [7ms]
    - Greater or equal of complex and complex [26ms]
    - Greater or equal of complex and number [10ms]
    - Greater or equal of number and complex [9ms]
```
e.g. functionally wise this PR offers enough for your to have `Big_Decimal` equality with other number types. Feel free to test it.

Reply:
Yes, it requires `from` conversion (not necessarily from `Number`, possibly just from `Integer`) & comparator. Both as [provided by complex number](https://github.com/enso-org/enso/pull/9070/files#diff-d92d98dfa353a1da9423b90033f0370b50ddc904d299d134b132fe05630865b2R30) sample. Moreover it is essential to have compatible [hash code implementation](https://github.com/enso-org/enso/pull/9070#discussion_r1491152185) between original & converted values.